src/core/ext/transport/chttp2/transport/chttp2_transport.c

```diff
@@ -712,6 +712,7 @@ static void destroy_stream_locked(grpc_exec_ctx *exec_ctx, void *sp,
 
   grpc_chttp2_list_remove_stalled_by_transport(t, s);
   grpc_chttp2_list_remove_stalled_by_stream(t, s);
+  grpc_chttp2_list_remove_writing_stream(t, s);
```

Comment:
in my humble opinion, the writing op and destroy_stream_locked both are scheduled by t->combiner  so there won't be race you mentioned, will there ?

Also where do you think it is better to do this remove ?